Trackers/centroid_tracker/centroid.py

```diff
@@ -52,28 +51,29 @@ def update(self, detections):
             - check if input centroids is less than last time, so some objects are lost, and icrease their miss and check if it >max_age deregister it.
                 - if not, then we new objects, so register them
         '''
-        if(detections==[] and self.tracks==[]):
-            return self.tracks
-        if(detections==[] and self.tracks!=[]):
-            print(id)
-            # increase every objects miss by 1
-            for track in self.tracks:
-                track.miss +=1
-            self.delete_track()
-            return self.tracks
 
-        inputCentroids = np.zeros((max(1, len(detections)), 2), dtype="int")
+        if(detections == []):
+            if(self.tracks == []):
+                return self.tracks
+            else:
+                # increase every objects miss by 1
+                for track in self.tracks:
+                    track.miss +=1
+                self.delete_track()
+                return self.tracks
+
+        inputCentroids = np.zeros((len(detections), 2), dtype="int")
 
-        for i in range(0,len(detections)):
-            cX = int((detections[i][2][0]+ detections[i][2][1]) / 2.0)
-            cY = int((detections[i][2][2] + detections[i][2][3]) / 2.0)
+        for i in range(len(detections)):
+            cX = int((detections[i][0]+ detections[i][1]) / 2.0)
+            cY = int((detections[i][2] + detections[i][3]) / 2.0)
             inputCentroids[i] = (cX, cY)
         
         objectId = []
         objectCentroid = []
         if len(self.tracks) == 0:
             for i in range(0, len(inputCentroids)):
-                self.add_track(self.nextID, detections,inputCentroids[i])
+                self.add_track(self.nextID, detections[i][2],inputCentroids[i])
```

Comment:
`detection[i][2]` here indicates the specific value present in the tuple. Here `bbox` is expected to be appended.

Reply:
yes, only `detections[i]` need to be given, as it contains bbox